### PR TITLE
[11.0][MIG] hr_contract

### PR DESCRIPTION
**Pull request by fayezqandeel**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_contract/migrations/11.0.1.0/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+hr_contract  / hr.contract              / state (selection)             : selection_keys is now '['cancel', 'close', 'draft', 'open', 'pending']' ('['close', 'draft', 'open', 'pending']')
+# Nothing to do, nothing related
+
+hr_contract  / hr.contract              / trial_date_start (date)       : DEL
+# Nothing to do, nothing related
+
+hr_contract  / hr.contract              / website_message_ids (one2many): NEW relation: mail.message
+# Nothing to do, new relations
+
+hr_contract  / hr.contract              / working_hours (many2one)      : DEL relation: resource.calendar
```
Comment (pedrobaeza) [inline addons/hr_contract/migrations/11.0.1.0/openupgrade_analysis_work.txt]: This is clearly a rename of the field from `working_hours` to `resource_calendar_id`, and more if you check the label from both fields in v10 and v11: `Working Schedule`. Please pay attention to these details.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by fayezqandeel** [comment]
@pedrobaeza rename field done on pre_migration.py

**Comment by pedrobaeza** [comment]
Completed the migration with:

* Fill company in contract from job.
* Remove deleted noupdate records.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
The unlink of the filter has revealed a possible problem with renamed_models method in OpenUpgrade. I'll investigate this further.

**Comment by pedrobaeza** [comment]
This depends indirectly of `base_automation`. We will probably need to add the dependency in the manifest for avoiding problems in the loading order (as we are unlinking base.automation records that are not anymore in this version, nor the dependency).

**Comment by pedrobaeza** [comment]
I just think about other better solution. Do the unlinking on end-migration where all the models are already loaded.

**Comment by pedrobaeza** [comment]
Correct after merging base_automation, so I merge.

**Pull request merged**